client/src/main/scala/grasshopper/client/addresspoints/AddressPointsClient.scala

```diff
@@ -30,7 +30,9 @@ object AddressPointsClient extends ServiceClient with AddressPointsJsonProtocol
 
   def geocode(address: String): Future[Either[ResponseError, AddressPointsResult]] = {
     implicit val ec: ExecutionContext = system.dispatcher
-    sendGetRequest(s"/addresses/points/${address}").flatMap { response =>
+    val addr = if (address.contains("?suggest=")) address else URLEncoder.encode(address, "UTF-8")
```

Comment:
I see a few issues here, which I think apply to the other service client classes as well:

1. I don't think we should allow upstream services to forward their URL fragments and params through to the  clients.  In general, each URL param (`suggest`) should correlate to a parameter on the underlying function (`geocode()`).  This also means any URL encode/decode logic would also be handled upstream as well.
1. We should be using a lib for parsing and building URIs.  As it stands now, the `suggest` logic wouldn't even get picked up if it wasn't the first param, and any additional params would be considered part of the address string.  Looks like Akka HTTP has a nice [`Uri`](http://doc.akka.io/api/akka-stream-and-http-experimental/1.0-RC4/index.html#akka.http.scaladsl.model.Uri) for building.  Not sure why it doesn't parsing too.  There's probably some other class for that. :)  
1. When this runs through a reverse proxy, the exact path may not be `/addesses/points/`.  It's probably worth adding a `basepath` to the `application.conf` so it can be overridden as an envvar.

None of this is a show-stopper.  If you can't get to this, perhaps we should spawn a "Standardize Service Clients" issue so we don't lose track of this.